We don't have PyNaCl and other supporting libraries on Solaris. Because of
that, we have to disable Ed25519 for now.

Original introduction pull request:
https://github.com/paramiko/paramiko/pull/972/files

--- paramiko-3.4.0/paramiko/__init__.py
+++ paramiko-3.4.0/paramiko/__init__.py
@@ -68,7 +68,6 @@ from paramiko.server import ServerInterf
 from paramiko.rsakey import RSAKey
 from paramiko.dsskey import DSSKey
 from paramiko.ecdsakey import ECDSAKey
-from paramiko.ed25519key import Ed25519Key
 from paramiko.sftp import SFTPError, BaseSFTP
 from paramiko.sftp_client import SFTP, SFTPClient
 from paramiko.sftp_server import SFTPServer
@@ -112,7 +111,7 @@ from paramiko.common import io_sleep
 
 
 # TODO: I guess a real plugin system might be nice for future expansion...
-key_classes = [DSSKey, RSAKey, Ed25519Key, ECDSAKey]
+key_classes = [DSSKey, RSAKey, ECDSAKey]
 
 
 __author__ = "Jeff Forcier <jeff@bitprophet.org>"
@@ -133,7 +132,6 @@ __all__ = [
     "CouldNotCanonicalize",
     "DSSKey",
     "ECDSAKey",
-    "Ed25519Key",
     "HostKeys",
     "Message",
     "MissingHostKeyPolicy",
--- paramiko-3.4.0/paramiko/client.py
+++ paramiko-3.4.0/paramiko/client.py
@@ -33,7 +33,6 @@ from paramiko.common import DEBUG
 from paramiko.config import SSH_PORT
 from paramiko.dsskey import DSSKey
 from paramiko.ecdsakey import ECDSAKey
-from paramiko.ed25519key import Ed25519Key
 from paramiko.hostkeys import HostKeys
 from paramiko.rsakey import RSAKey
 from paramiko.ssh_exception import (
@@ -725,7 +724,7 @@ class SSHClient(ClosingContextManager):
             for key_filename in key_filenames:
                 # TODO 4.0: leverage PKey.from_path() if we don't end up just
                 # killing SSHClient entirely
-                for pkey_class in (RSAKey, DSSKey, ECDSAKey, Ed25519Key):
+                for pkey_class in (RSAKey, DSSKey, ECDSAKey):
                     try:
                         key = self._key_from_filepath(
                             key_filename, pkey_class, passphrase
@@ -767,7 +766,6 @@ class SSHClient(ClosingContextManager):
                 (RSAKey, "rsa"),
                 (DSSKey, "dsa"),
                 (ECDSAKey, "ecdsa"),
-                (Ed25519Key, "ed25519"),
             ]:
                 # ~/ssh/ is for windows
                 for directory in [".ssh", "ssh"]:
--- paramiko-3.4.0/paramiko/pkey.py
+++ paramiko-3.4.0/paramiko/pkey.py
@@ -128,7 +128,7 @@ class PKey:
         # TODO: make sure sphinx is reading Path right in param list...
 
         # Lazy import to avoid circular import issues
-        from paramiko import DSSKey, RSAKey, Ed25519Key, ECDSAKey
+        from paramiko import DSSKey, RSAKey, ECDSAKey
 
         # Normalize to string, as cert suffix isn't quite an extension, so
         # pathlib isn't useful for this.
@@ -171,8 +171,6 @@ class PKey:
             key_class = DSSKey
         elif isinstance(loaded, asymmetric.rsa.RSAPrivateKey):
             key_class = RSAKey
-        elif isinstance(loaded, asymmetric.ed25519.Ed25519PrivateKey):
-            key_class = Ed25519Key
         elif isinstance(loaded, asymmetric.ec.EllipticCurvePrivateKey):
             key_class = ECDSAKey
         else:
--- paramiko-3.4.0/paramiko/transport.py
+++ paramiko-3.4.0/paramiko/transport.py
@@ -91,7 +91,6 @@ from paramiko.common import (
 )
 from paramiko.compress import ZlibCompressor, ZlibDecompressor
 from paramiko.dsskey import DSSKey
-from paramiko.ed25519key import Ed25519Key
 from paramiko.kex_curve25519 import KexCurve25519
 from paramiko.kex_gex import KexGex, KexGexSHA256
 from paramiko.kex_group1 import KexGroup1
@@ -178,7 +177,6 @@ class Transport(threading.Thread, Closin
     )
     # ~= HostKeyAlgorithms in OpenSSH land
     _preferred_keys = (
-        "ssh-ed25519",
         "ecdsa-sha2-nistp256",
         "ecdsa-sha2-nistp384",
         "ecdsa-sha2-nistp521",
@@ -189,7 +187,6 @@ class Transport(threading.Thread, Closin
     )
     # ~= PubKeyAcceptedAlgorithms
     _preferred_pubkeys = (
-        "ssh-ed25519",
         "ecdsa-sha2-nistp256",
         "ecdsa-sha2-nistp384",
         "ecdsa-sha2-nistp521",
@@ -293,8 +290,6 @@ class Transport(threading.Thread, Closin
         "ecdsa-sha2-nistp384-cert-v01@openssh.com": ECDSAKey,
         "ecdsa-sha2-nistp521": ECDSAKey,
         "ecdsa-sha2-nistp521-cert-v01@openssh.com": ECDSAKey,
-        "ssh-ed25519": Ed25519Key,
-        "ssh-ed25519-cert-v01@openssh.com": Ed25519Key,
     }
 
     _kex_info = {
--- paramiko-3.4.0/tests/conftest.py
+++ paramiko-3.4.0/tests/conftest.py
@@ -13,7 +13,6 @@ from paramiko import (
     Transport,
     DSSKey,
     RSAKey,
-    Ed25519Key,
     ECDSAKey,
     PKey,
 )
@@ -122,11 +121,6 @@ key_data = [
 key_data = [
     ["ssh-rsa", RSAKey, "SHA256:OhNL391d/beeFnxxg18AwWVYTAHww+D4djEE7Co0Yng"],
     ["ssh-dss", DSSKey, "SHA256:uHwwykG099f4M4kfzvFpKCTino0/P03DRbAidpAmPm0"],
-    [
-        "ssh-ed25519",
-        Ed25519Key,
-        "SHA256:J6VESFdD3xSChn8y9PzWzeF+1tl892mOy2TqkMLO4ow",
-    ],
     [
         "ecdsa-sha2-nistp256",
         ECDSAKey,
--- paramiko-3.4.0/tests/pkey.py
+++ paramiko-3.4.0/tests/pkey.py
@@ -7,7 +7,6 @@ from cryptography.hazmat.primitives.asym
 from paramiko import (
     DSSKey,
     ECDSAKey,
-    Ed25519Key,
     Message,
     PKey,
     PublicBlob,
@@ -156,12 +155,6 @@ class PKey_:
             msg.rewind()
             assert key.verify_ssh_sig(b"ice weasels", msg)  # our data verified
 
-        def loading_cert_of_different_type_from_key_raises_ValueError(self):
-            edkey = Ed25519Key.from_private_key_file(_support("ed25519.key"))
-            err = "PublicBlob type ssh-rsa-cert-v01@openssh.com incompatible with key type ssh-ed25519"  # noqa
-            with raises(ValueError, match=err):
-                edkey.load_certificate(_support("rsa.key-cert.pub"))
-
     def fingerprint(self, keys):
         # NOTE: Hardcoded fingerprint expectation stored in fixture.
         assert keys.pkey.fingerprint == keys.expected_fp
@@ -174,8 +167,6 @@ class PKey_:
             assert key.algorithm_name == "DSS"
         elif isinstance(key, ECDSAKey):
             assert key.algorithm_name == "ECDSA"
-        elif isinstance(key, Ed25519Key):
-            assert key.algorithm_name == "ED25519"
         # TODO: corner case: AgentKey, whose .name can be cert-y (due to the
         # value of the name field passed via agent protocol) and thus
         # algorithm_name is eg "RSA-CERT" - keys loaded directly from disk will
@@ -218,9 +209,6 @@ class PKey_:
         def dss_is_protocol_name(self):
             assert DSSKey.identifiers() == ["ssh-dss"]
 
-        def ed25519_is_protocol_name(self):
-            assert Ed25519Key.identifiers() == ["ssh-ed25519"]
-
         def ecdsa_is_all_curve_names(self):
             assert ECDSAKey.identifiers() == [
                 "ecdsa-sha2-nistp256",
--- paramiko-3.4.0/tests/test_client.py
+++ paramiko-3.4.0/tests/test_client.py
@@ -52,7 +52,6 @@ FINGERPRINTS = {
     "ssh-dss": b"\x44\x78\xf0\xb9\xa2\x3c\xc5\x18\x20\x09\xff\x75\x5b\xc1\xd2\x6c",  # noqa
     "ssh-rsa": b"\x60\x73\x38\x44\xcb\x51\x86\x65\x7f\xde\xda\xa2\x2b\x5a\x57\xd5",  # noqa
     "ecdsa-sha2-nistp256": b"\x25\x19\xeb\x55\xe6\xa1\x47\xff\x4f\x38\xd2\x75\x6f\xa5\xd5\x60",  # noqa
-    "ssh-ed25519": b'\xb3\xd5"\xaa\xf9u^\xe8\xcd\x0e\xea\x02\xb9)\xa2\x80',
 }
 
 
@@ -271,10 +270,6 @@ class SSHClientTest(ClientTest):
         self._test_connection(key_filename=_support("ecdsa-256.key"))
 
     @requires_sha1_signing
-    def test_client_ed25519(self):
-        self._test_connection(key_filename=_support("ed25519.key"))
-
-    @requires_sha1_signing
     def test_multiple_key_files(self):
         """
         verify that SSHClient accepts and tries multiple key files.
@@ -327,7 +322,7 @@ class SSHClientTest(ClientTest):
         # They're similar except for which path is given; the expected auth and
         # server-side behavior is 100% identical.)
         # NOTE: only bothered whipping up one cert per overall class/family.
-        for type_ in ("rsa", "dss", "ecdsa-256", "ed25519"):
+        for type_ in ("rsa", "dss", "ecdsa-256"):
             key_path = _support(f"{type_}.key")
             self._test_connection(
                 key_filename=key_path,
@@ -342,7 +337,7 @@ class SSHClientTest(ClientTest):
         # about the server-side key object's public blob. Thus, we can prove
         # that a specific cert was found, along with regular authorization
         # succeeding proving that the overall flow works.
-        for type_ in ("rsa", "dss", "ecdsa-256", "ed25519"):
+        for type_ in ("rsa", "dss", "ecdsa-256"):
             key_path = _support(f"{type_}.key")
             self._test_connection(
                 key_filename=key_path,
--- paramiko-3.4.0/tests/test_hostkeys.py
+++ paramiko-3.4.0/tests/test_hostkeys.py
@@ -36,8 +36,6 @@ broken.example.com ssh-rsa AAAA
 happy.example.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAIEA8bP1ZA7DCZDB9J0s50l31M\
 BGQ3GQ/Fc7SX6gkpXkwcZryoi4kNFhHu5LvHcZPdxXV1D+uTMfGS1eyd2Yz/DoNWXNAl8TI0cAsW\
 5ymME3bQ4J/k1IKxCtz/bAlAqFgKoc+EolMziDYqWIATtW0rYTJvzGAzTmMj80/QpsFH+Pc2M=
-modern.example.com ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIKHEChAIxsh2hr8Q\
-+Ea1AAHZyfEB2elEc2YgduVzBtp+
 curvy.example.com ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlz\
 dHAyNTYAAABBBAa+pY7djSpbg5viAcZhPt56AO3U3Sd7h7dnlUp0EjfDgyYHYQxl2QZ4JGgfwR5iv9\
 T9iRZjQzvJd5s+kBAZtpk=
@@ -50,8 +48,6 @@ D+jrpI9cycZHqilK0HmxDeCuxbwyMuaCygU9gS2q
 happy.example.com\tssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAIEA8bP1ZA7DCZDB9J0s50l31M\
 BGQ3GQ/Fc7SX6gkpXkwcZryoi4kNFhHu5LvHcZPdxXV1D+uTMfGS1eyd2Yz/DoNWXNAl8TI0cAsW\
 5ymME3bQ4J/k1IKxCtz/bAlAqFgKoc+EolMziDYqWIATtW0rYTJvzGAzTmMj80/QpsFH+Pc2M=
-modern.example.com\tssh-ed25519\tAAAAC3NzaC1lZDI1NTE5AAAAIKHEChAIxsh2hr8Q\
-+Ea1AAHZyfEB2elEc2YgduVzBtp+
 curvy.example.com\tecdsa-sha2-nistp256\tAAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbml\
 zdHAyNTYAAABBBAa+pY7djSpbg5viAcZhPt56AO3U3Sd7h7dnlUp0EjfDgyYHYQxl2QZ4JGgfwR5iv\
 9T9iRZjQzvJd5s+kBAZtpk=
@@ -83,7 +79,7 @@ class HostKeysTest(unittest.TestCase):
 
     def test_load(self):
         hostdict = paramiko.HostKeys("hostfile.temp")
-        assert len(hostdict) == 4
+        assert len(hostdict) == 3
         self.assertEqual(1, len(list(hostdict.values())[0]))
         self.assertEqual(1, len(list(hostdict.values())[1]))
         fp = hexlify(
@@ -96,7 +92,7 @@ class HostKeysTest(unittest.TestCase):
         hh = "|1|BMsIC6cUIP2zBuXR3t2LRcJYjzM=|hpkJMysjTk/+zzUUzxQEa2ieq6c="
         key = paramiko.RSAKey(data=decodebytes(keyblob))
         hostdict.add(hh, "ssh-rsa", key)
-        assert len(hostdict) == 5
+        assert len(hostdict) == 4
         x = hostdict["foo.example.com"]
         fp = hexlify(x["ssh-rsa"].get_fingerprint()).upper()
         self.assertEqual(b"7EC91BB336CB6D810B124B1353C32396", fp)
@@ -113,7 +109,7 @@ class HostKeysTest(unittest.TestCase):
         fp = hexlify(x["ssh-rsa"].get_fingerprint()).upper()
         self.assertEqual(b"E6684DB30E109B67B70FF1DC5C7F1363", fp)
         assert list(hostdict) == hostdict.keys()
-        assert len(list(hostdict)) == len(hostdict.keys()) == 4
+        assert len(list(hostdict)) == len(hostdict.keys()) == 3
 
     def test_dict_set(self):
         hostdict = paramiko.HostKeys("hostfile.temp")
@@ -123,7 +119,7 @@ class HostKeysTest(unittest.TestCase):
         hostdict["fake.example.com"] = {}
         hostdict["fake.example.com"]["ssh-rsa"] = key
 
-        assert len(hostdict) == 5
+        assert len(hostdict) == 4
         self.assertEqual(2, len(list(hostdict.values())[0]))
         self.assertEqual(1, len(list(hostdict.values())[1]))
         self.assertEqual(1, len(list(hostdict.values())[2]))
--- paramiko-3.4.0/tests/test_pkey.py
+++ paramiko-3.4.0/tests/test_pkey.py
@@ -32,7 +32,6 @@ from paramiko import (
     RSAKey,
     DSSKey,
     ECDSAKey,
-    Ed25519Key,
     Message,
     util,
     SSHException,
@@ -567,65 +566,6 @@ class KeyTest(unittest.TestCase):
         comparable = TEST_KEY_BYTESTR
         self.assertEqual(str(key), comparable)
 
-    def test_ed25519(self):
-        key1 = Ed25519Key.from_private_key_file(_support("ed25519.key"))
-        key2 = Ed25519Key.from_private_key_file(
-            _support("test_ed25519_password.key"), b"abc123"
-        )
-        self.assertNotEqual(key1.asbytes(), key2.asbytes())
-
-    def test_ed25519_funky_padding(self):
-        # Proves #1306 by just not exploding with 'Invalid key'.
-        Ed25519Key.from_private_key_file(
-            _support("test_ed25519-funky-padding.key")
-        )
-
-    def test_ed25519_funky_padding_with_passphrase(self):
-        # Proves #1306 by just not exploding with 'Invalid key'.
-        Ed25519Key.from_private_key_file(
-            _support("test_ed25519-funky-padding_password.key"), b"asdf"
-        )
-
-    def test_ed25519_compare(self):
-        # verify that the private & public keys compare equal
-        key = Ed25519Key.from_private_key_file(_support("ed25519.key"))
-        self.assertEqual(key, key)
-        pub = Ed25519Key(data=key.asbytes())
-        self.assertTrue(key.can_sign())
-        self.assertTrue(not pub.can_sign())
-        self.assertEqual(key, pub)
-
-    # No point testing on systems that never exhibited the bug originally
-    @pytest.mark.skipif(
-        not is_low_entropy(), reason="Not a low-entropy system"
-    )
-    def test_ed25519_32bit_collision(self):
-        # Re: 2021.10.19 security report email: two different private keys
-        # which Paramiko compared as equal on low-entropy platforms.
-        original = Ed25519Key.from_private_key_file(
-            _support("badhash_key1.ed25519.key")
-        )
-        generated = Ed25519Key.from_private_key_file(
-            _support("badhash_key2.ed25519.key")
-        )
-        assert original != generated
-
-    def test_ed25519_nonbytes_password(self):
-        # https://github.com/paramiko/paramiko/issues/1039
-        Ed25519Key.from_private_key_file(
-            _support("test_ed25519_password.key"),
-            # NOTE: not a bytes. Amusingly, the test above for same key DOES
-            # explicitly cast to bytes...code smell!
-            "abc123",
-        )
-        # No exception -> it's good. Meh.
-
-    def test_ed25519_load_from_file_obj(self):
-        with open(_support("ed25519.key")) as pkey_fileobj:
-            key = Ed25519Key.from_private_key(pkey_fileobj)
-        self.assertEqual(key, key)
-        self.assertTrue(key.can_sign())
-
     def test_keyfile_is_actually_encrypted(self):
         # Read an existing encrypted private key
         file_ = _support("test_rsa_password.key")
--- paramiko-3.4.0/tests/test_transport.py
+++ paramiko-3.4.0/tests/test_transport.py
@@ -1244,7 +1244,7 @@ class TestExtInfo(unittest.TestCase):
             # data stored on Transport after hearing back from a compatible
             # server (such as ourselves in server mode)
             assert tc.server_extensions == {
-                "server-sig-algs": b"ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,rsa-sha2-512,rsa-sha2-256,ssh-rsa,ssh-dss"  # noqa
+                "server-sig-algs": b"ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,rsa-sha2-512,rsa-sha2-256,ssh-rsa,ssh-dss"  # noqa
             }
 
     def test_client_uses_server_sig_algs_for_pubkey_auth(self):
--- paramiko-3.4.0/setup.py
+++ paramiko-3.4.0/setup.py
@@ -89,7 +89,6 @@ setup(
     install_requires=[
         "bcrypt>=3.2",
         "cryptography>=3.3",
-        "pynacl>=1.5",
     ],
     extras_require=extras_require,
 )
--- paramiko-3.4.0/paramiko.egg-info/requires.txt
+++ paramiko-3.4.0/paramiko.egg-info/requires.txt
@@ -1,6 +1,5 @@
 bcrypt>=3.2
 cryptography>=3.3
-pynacl>=1.5
 
 [all]
 pyasn1>=0.1.7
